subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftLibraryPlugin.java

```diff
@@ -0,0 +1,122 @@
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.ConfigurablePublishArtifact;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.attributes.Usage;
+import org.gradle.api.file.ConfigurableFileTree;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.language.base.plugins.LifecycleBasePlugin;
+import org.gradle.language.cpp.plugins.CppBasePlugin;
+import org.gradle.language.swift.tasks.SwiftCompile;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+import org.gradle.nativeplatform.toolchain.NativeToolChain;
+import org.gradle.nativeplatform.toolchain.internal.NativeToolChainInternal;
+import org.gradle.nativeplatform.toolchain.internal.NativeToolChainRegistryInternal;
+
+import java.util.Collections;
+
+/**
+ * <p>A plugin that produces a native library from Swift source.</p>
+ *
+ * <p>Assumes the source files are located in `src/main/swift`.</p>
+ *
+ * @since 4.1
+ */
+public class SwiftLibraryPlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply(SwiftBasePlugin.class);
+
+        // TODO - extract some common code to setup the compile task and conventions
+        // Add a compile task
+        final SwiftCompile compile = project.getTasks().create("compileSwift", SwiftCompile.class);
+
+        compile.includes(project.getConfigurations().getByName(SwiftBasePlugin.SWIFT_IMPORT_PATH));
+        compile.lib(project.getConfigurations().getByName(CppBasePlugin.NATIVE_LINK));
+
+        ConfigurableFileTree sourceTree = project.fileTree("src/main/swift");
+        sourceTree.include("**/*.swift");
+        compile.source(sourceTree);
+
+        // TODO - should use PIC
+        compile.setCompilerArgs(Lists.newArrayList("-emit-library"));
+        compile.setMacros(Collections.<String, String>emptyMap());
+        compile.setModuleName(project.getName());
+
+        // TODO - should reflect changes to build directory
+        compile.setObjectFileDir(project.file("build/main/objs"));
+
+        DefaultNativePlatform currentPlatform = new DefaultNativePlatform("current");
+        compile.setTargetPlatform(currentPlatform);
+
+        // TODO - make this lazy
+        NativeToolChain toolChain = ((ProjectInternal) project).getModelRegistry().realize("toolChains", NativeToolChainRegistryInternal.class).getForPlatform(currentPlatform);
+        compile.setToolChain(toolChain);
+
+        String libName = ((NativeToolChainInternal) toolChain).select(currentPlatform).getSharedLibraryName("build/lib/" + project.getName());
+        compile.setOutputFile(project.file(libName));
+
+        project.getTasks().getByName(LifecycleBasePlugin.ASSEMBLE_TASK_NAME).dependsOn(compile);
+
+        // TODO - add lifecycle tasks
+
+
+        Configuration apiElements = project.getConfigurations().create("swiftApiElements");
```

Comment:
The dependency management was heavily inspired by C++. Please give a good review to ensure we don't make any obvious mistake.